test/OpenFeature.E2ETests/Steps/MetadataStepDefinitions.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OpenFeature.Model;
+using OpenFeature.Providers.Memory;
+using Reqnroll;
+using Xunit;
+
+namespace OpenFeature.E2ETests.Steps;
+
+[Binding]
+[Scope(Feature = "Metadata")]
+public class MetadataStepDefinitions
+{
+    private object _objResult = null!;
+
+    private string _flagKey = null!;
+    private string _defaultValue = null!;
+    private FlagType? _flagType;
+
+    private FeatureClient? _client;
+    private static readonly IDictionary<string, Flag> E2EFlagConfig = new Dictionary<string, Flag>
+    {
+        {
+            "metadata-flag", new Flag<bool>(
+                variants: new Dictionary<string, bool> { { "on", true }, { "off", false } },
+                defaultVariant: "on",
+                flagMetadata: new ImmutableMetadata(new Dictionary<string, object>
+                {
+                    { "string", "1.0.2" }, { "integer", 2 }, { "float", 0.1 }, { "boolean", true }
+                })
+            )
+        },
+        {
+            "boolean-flag", new Flag<bool>(
+                variants: new Dictionary<string, bool> { { "on", true }, { "off", false } },
+                defaultVariant: "on"
+            )
+        },
+        {
+            "integer-flag", new Flag<int>(
+                variants: new Dictionary<string, int> { { "23", 23 }, { "42", 42 } },
+                defaultVariant: "23"
+            )
+        },
+        {
+            "float-flag", new Flag<double>(
+                variants: new Dictionary<string, double> { { "2.3", 2.3 }, { "4.2", 4.2 } },
+                defaultVariant: "2.3"
+            )
+        },
+        {
+            "string-flag", new Flag<string>(
+                variants: new Dictionary<string, string> { { "value", "value" }, { "value2", "value2" } },
+                defaultVariant: "value"
+            )
+        }
+    };
+
+    [Given("a stable provider")]
+    public void GivenAStableProvider()
```

Comment:
you define this step multiple times, but gherkin/cucumber normally can use multiple step classes together. so you could group all for metadata in a metadataStep class, all for provider in a provider step class - https://github.com/open-feature/java-sdk/tree/main/src/test/java/dev/openfeature/sdk/e2e/steps is the way how we structure them in java.

The big benefit of this approach is, as soon as one step is defined properly, we can generate new tests out of the well defined steps, by just writing gherkin feature files. We utilized this already heavily for flagd, and it helped us to write fast and easy tests, without actually touching code.